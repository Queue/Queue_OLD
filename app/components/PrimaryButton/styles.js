//
// PrimaryButton styles.js

import { StyleSheet } from 'react-native';
import Colors from '../../lib/colors';

export default styles = StyleSheet.create({
  primaryBtn: {
    fontSize: 20,
    fontWeight: "500"
  },
  btnContainer: {
    padding: 8,
    margin: 15,
    marginTop: 20,
    overflow:'hidden',
    borderWidth: 1,
    borderColor: Colors.primaryForeground,
    borderRadius: 4
  }
});
