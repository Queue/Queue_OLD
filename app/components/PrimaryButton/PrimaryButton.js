//
// Btn component file

import React from 'react'
import styles from './styles';

// Custom simple button with wrapper
import Button from 'react-native-button';
import Colors from '../../lib/colors';

export default PrimaryBtn = ({name, press, buttonColor, fontSize}) => {

  styles.buttonOverwrite = {
    color: buttonColor,
    fontSize: fontSize
  }

  styles.containerOverwrite = {
    borderColor: buttonColor
  }

  return (
    <Button
      onPress = {press}
      style = {[styles.primaryBtn, styles.buttonOverwrite]}
      containerStyle = {[styles.btnContainer, styles.containerOverwrite]}>
      {name}
    </Button>
  );
};

PrimaryBtn.propType = {
  name: React.PropTypes.string,
  press: React.PropTypes.func,
  buttonColor: React.PropTypes.string,
  fontSize: React.PropTypes.number
}

PrimaryBtn.defaultProps = {
  buttonColor: Colors.primaryForeground,
  fontSize: 20
};
