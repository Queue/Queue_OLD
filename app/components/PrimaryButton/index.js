//
// PrimaryButton index.js

import PrimaryButton from './PrimaryButton';
export default PrimaryButton;
