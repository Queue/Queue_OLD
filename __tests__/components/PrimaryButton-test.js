//
// PrimaryButton component test

import React from 'react';
import PrimaryButton from '../../app/components/PrimaryButton';
import renderer from 'react-test-renderer';

let testFunc = () => {
  console.warn('');
};

it('renders correctly', () => {
  const tree = renderer.create(
    <PrimaryButton
      press={testFunc}
      name={'Log In'}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
