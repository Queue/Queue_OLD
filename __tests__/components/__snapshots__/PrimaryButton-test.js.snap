exports[`test renders correctly 1`] = `
<View
  accessibilityComponentType="button"
  accessibilityLabel={undefined}
  accessibilityTraits="button"
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderColor": "#4AC948",
      "borderRadius": 4,
      "borderWidth": 1,
      "margin": 15,
      "marginTop": 20,
      "opacity": 1,
      "overflow": "hidden",
      "padding": 8,
    }
  }
  testID={undefined}>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#007aff",
          "fontFamily": "HelveticaNeue-Medium",
          "fontSize": 17,
          "fontWeight": "bold",
          "textAlign": "center",
        },
        null,
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "500",
          },
          Object {
            "color": "#4AC948",
          },
        ],
        null,
      ]
    }>
    Log In
  </Text>
</View>
`;
